src/App.js

```diff
@@ -1,9 +1,149 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import Header from './components/Header';
+import Main from './components/Main';
+import Modal from './components/Modal';
+import ConfirmModal from './components/ConfirmModal';
+import styled, { createGlobalStyle } from 'styled-components';
+
+const GlobalStyle = createGlobalStyle`
+  * {
+    margin: 0;
+    box-sizing: border-box;
+    font-family: 'Pretendard Variable', sans-serif;
+  }
+  @font-face {
+    font-family: 'Pretendard Variable';
+    font-weight: 45 920;
+    font-style: normal;
+    font-display: swap;
+    src: url('/fonts/PretendardVariable.woff2') format('woff2-variations');
+  }
+`; // 전체 페이지의 기본 여백 제거, 코드의 중복을 줄이고 모든 컴포넌트에 일관된 스타일 적용
+
+const AppContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
 function App() {
+  const [todos, setTodos] = useState([]);
+  const [isAllCompleted, setIsAllCompleted] = useState(false); // 모든 todo가 완료되었는지 여부를 추적
+  const [modalText, setModalText] = useState(''); // 기능마다 모달 text를 다르게 하기 위함
+  const [isModalOpen, setIsModalOpen] = useState(false); // 모달 상태
+  const [isConfirmModalOpen, setIsConfirmModalOpen] = useState(false);
+  const [todoToDelete, setTodoToDelete] = useState(null); // 삭제할 todo
+
+  // 전체 todo와 완료된 todo의 개수를 계산하는 함수
+  const totalTodos = todos.length;
+  const completedTodos = todos.filter(todo => todo.completed).length;
+
+
+  useEffect(() => {
+    const storedTodos = localStorage.getItem('todos');
+    if (storedTodos) {
+      try {
+        setTodos(JSON.parse(storedTodos)); // 데이터가 있다면 JSON 파싱
+      } catch (error) {
+        console.error("로컬 스토리지에서 데이터를 불러오는 중 오류 발생:", error);
+        setTodos([]); // 오류가 있으면 빈 배열로 초기화
+      }
+    }
+    const storedIsAllCompleted = localStorage.getItem('isAllCompleted');
+    if (storedIsAllCompleted) {
+      setIsAllCompleted(JSON.parse(storedIsAllCompleted));
+    }
+  }, []); // 새로고침 시 이전 데이터를 불러오기. localStorage로부터 todo 배열을 todo state에 저장하고 모든 todo 완료 상태를 저장
```

Comment:
```suggestion

    if (localStorage.getItem('isAllCompleted')) {
      setIsAllCompleted(JSON.parse(localStorage.getItem('isAllCompleted')));
    }
  }, []); // 새로고침 시 이전 데이터를 불러오기. localStorage로부터 todo 배열을 todo state에 저장하고 모든 todo 완료 상태를 저장
```
그냥 깔끔하게 이렇게도 쓸 수 있겠네요!